draft-thomassen-dnsop-multialgo.md

```diff
@@ -250,21 +275,26 @@ use the INSECURE algorithm, it will continue to fully validate with
 supporting resolvers, while non-supporting resolvers will treat the
 zone as insecure until the algorithm is replaced.
 
-Conversely, when an algorithm is added to the set of UNIVERSAL ones,
-it is conceivable that a signer may move to this algorithm before all
-validators are upgraded.  This is, in fact, not a problem, as
-resolvers do not need to know the concept of UNIVERSAL.  A problem
-could only occur if the corresponding RRSIG was not supported by the
-resolver; however, in that case labeling the algorithm as UNIVERSAL would
-have been premature.  Determining universal support cannot be solved on
-the protocol level, and it is the community's responsibility to only
-advance an algorithm to UNIVERSAL if safe enough, i.e. if the
-number of resolvers lacking support is deemed negligible.
-
-In any case, regardless of "who moves first", resolution is never
+Conversely, when an algorithm is added to the set of UNIVERSAL ones, signers
+MAY begin to return signatures for just that algorithm.  This is, in fact, not
+a problem, as resolvers do not need to know the concept of UNIVERSAL, they just
+need to support that algorithm (or in less typically explicitly classify it as
+INSECURE).  A problem could only occur if the corresponding RRSIG was not
+supported by a non-negligible population of resolvers; however, in that case
+labeling the algorithm as UNIVERSAL would have been premature.  Determining
+universal support cannot be solved on the protocol level, and it is the
+community's responsibility to only advance an algorithm to UNIVERSAL when safe
+enough, i.e. when the population of resolvers lacking support is deemed negligible.
+
+<!-- In any case, regardless of "who moves first", resolution is never
 disrupted, and changes to the set of UNIVERSAL algorithms do not
 trigger overly conservative SERVFAIL responses.
 
+Viktor:  This is not true.  If algorithms are prematurely deemed UNIVERSAL by
+signers, and are not yet universall supported by resolvers, problems are sure
+to happen.
+-->
```

Comment:
Oh yes -- it's only true when "playing by the rules", i.e. not prematurely labeling any algorithms UNIVERSAL.

I think it's worthwhile noting that when nothing premature is going on, resolution will not be disrupted. Do you have a suggestion how to make that point?

Reply:
The point you can make is that resolvers don't need to pay attention to which/when algorithms are UNIVERSAL, they only need to mark algorithms they've stopped supporting as "INSECURE".

Signers on the other hand do need to know which algorithms are UNIVERSAL and can only skip supplying all the RRSIGs when vending only a currently or previously UNIVERSAL signature (long established, but now possibly disabled at some resolvers).  That is recognised by all resolvers and either supported or considered INSECURE.

This means I think that signers DO need to also know which algorithms fall into the second category.  Otherwise a signer has no basis to believe that it is OK to publish only 7 and not 13 when both are in the DS and DNSKEY RRsets (but not necessarily in RRSIGs).

Reply:
So there still seems to be a gap in the design.  The point being essentially that an algorithm that was previously universally supported is still universally "understood" by resolvers that adopt the new criteria, and they MUST either accept it or treat as "INSECURE" rather than merely unsupported (like with a "novel" algorithm).

Once an algorithm reaches broad recognition, it can be published in isolation.  Novel algorithms must be accompanied by all the other novel algorithms or at least one universal RRSIG.

This probably needs a bit more text.  Do you see what I'm getting at?

Reply:
Yes, but I can't think of good text for this right now, and I'd like to move forward with the document merge asap so that we don't have to maintain two copies in separate repos. For now, I removed the paragraph in question, and I'll open an issue for addressing this point in the other repo.

Reply:
https://github.com/shuque/draft-dnsop-multi-alg-rules/issues/4